src/input-nested-select/input-nested-select.tsx

```diff
@@ -74,10 +74,12 @@ export const InputNestedSelect = <V1, V2, V3>({
     // EFFECTS
     // =============================================================================
     useEffect(() => {
-        const newKeyPath = _selectedKeyPath || [];
-        setSelectedKeyPath(newKeyPath);
+        const newKeyPath = _selectedKeyPath?.flat().length
+            ? [_selectedKeyPath]
+            : [];
```

Comment:
can I confirm that the desired behaviour is:

if `selectedKeyPath` is
- undefined, init to `[]`
- empty array, init to `[]`
- non-empty, init to `[selectedKeyPath]`

this can be simplified to:
```suggestion
        const newKeyPath = _selectedKeyPath?.length ? [_selectedKeyPath] : [];
```

I don't think `flat` does anything here because it's already a 1-dimensional array